protocol-units/da/m1/util/src/ir_blob.rs

```diff
@@ -24,6 +24,20 @@ impl InnerSignedBlobV1Data {
 		Self { blob, timestamp }
 	}
 
+	pub fn compute_id<C>(&self) -> Vec<u8>
```

Comment:
There should be a doc comment explaining the purpose of the method.

Reply:
If we assume that the IR blob format bakes in the hash algorithm for these IDs, the ID could be a newtype wrapping a fixed-size array and exposing the array length as an associated item.

I haven't tried to reconcile const generics with `ArrayLength`, so it might be more difficult than it sounds. But please for the love of everything, do not use `generic-array` in APIs more than it already is.

Reply:
Will add and fix.